src/main/java/org/testng/reporters/XMLSuiteResultWriter.java

```diff
@@ -284,23 +284,39 @@ private void addTestResultException(XMLStringBuffer xmlBuffer, ITestResult testR
       }
 
       String[] stackTraces = Utils.stackTrace(exception, false);
```

Comment:
Perf/improvment: if the stacktrace is not displayed, `Utils.stackTrace` should not be computed.